### PR TITLE
OCPBUGS-3469: Multi-arch support

### PR DESCRIPTION
**Pull request by oceanc80**
Add configurable multi-arch build support

### CODE REVIEW

File: pkg/quarkus/v1alpha/scaffolds/internal/templates/makefile.go [hunk 1]
```diff
@@ -62,8 +62,11 @@ func (f *Makefile) SetTemplateDefaults() error {
 }
 
 const makefileTemplate = `
-# Image URL to use all building/pushing image targets
+# Image URL to use for all building/pushing image targets
 IMG ?= {{ .Image }}
+# Operating system and architecture to use for building image
+OS ?= {{ .OS }}
+ARCH ?= {{ .Arch }}
```
Comment (everettraven) [inline pkg/quarkus/v1alpha/scaffolds/internal/templates/makefile.go:69]: One thing to note here is that with the Go templates, to use the notation `{{ .X }}` the field has to exist on the object being used to render the template. When I attempted to run `operator-sdk init ...` with the changes on this branch I received the error:
```
FATA[0000] failed to initialize project: unable to scaffold with "quarkus.javaoperatorsdk.io/v1-alpha": template: *templates.Makefile:5:9: executing "*templates.Makefile" at <.OS>: can't evaluate field OS in type *templates.Makefile 
```

To fix this I think we need to:
1. Update the `Makefile` struct definition:
https://github.com/operator-framework/java-operator-plugins/blob/0eed476d2b7548daf184fb7a836ab8ac8e0dead9/pkg/quarkus/v1alpha/scaffolds/internal/templates/makefile.go#L25-L37
to have those fields.

2. Update the `Makefile.SetTemplateDefaults()` function:
https://github.com/operator-framework/java-operator-plugins/blob/0eed476d2b7548daf184fb7a836ab8ac8e0dead9/pkg/quarkus/v1alpha/scaffolds/internal/templates/makefile.go#L39-L62
to set the default values if they are not specified. I think a good default here would be `OS := linux` && `Arch := amd64`.

**Review by everettraven — CHANGES_REQUESTED** [review]
These changes currently don't work when attempting to initialize a project using `operator-sdk init ...`. I left a comment showing the error I received and what I think the solution is.

Another thing I noticed is there is a few commits with multiple contributors and looks like it would re-commit a few already merged commits. I think a `git rebase` (or similar operation) may have gone a bit off the rails as I would have only expected the Makefile template and the testdata to be updated by this PR.

**Review by laxmikantbpandhare — APPROVED** [review]
/lgtm

I will try updated tutorial steps on the weekend.

**Comment by everettraven** [comment]
So I think the changes look good and make sense, but I just can't seem to get an image to actually be built when running `make docker-build`. However, this isn't unique to this PR and also happens when trying to run `make docker-build` on the testdata/quarkus/memcached-quarkus-operator sample on the main branch.

**Review by everettraven — APPROVED** [review]
/lgtm

**Pull request merged**